pkg/namespace/builder.go

```diff
@@ -115,6 +128,65 @@ func AllowedRelationWithExpiration(namespaceName string, relationName string) *c
 	}
 }
 
+// AllowedRelationWithDeprecation creates a relation reference to an allowed relation.
+func AllowedRelationWithDeprecation(namespaceName string, relationName string, deprecation *core.Deprecation) *core.AllowedRelation {
+	return &core.AllowedRelation{
+		Namespace: namespaceName,
+		RelationOrWildcard: &core.AllowedRelation_Relation{
+			Relation: relationName,
+		},
+		Deprecation: deprecation,
+	}
+}
+
+// AllowedRelationWithDeprecationAndCaveat creates a relation reference to an allowed relation.
+func AllowedRelationWithDeprecationAndCaveat(namespaceName string, relationName string, deprecation *core.Deprecation, withCaveat *core.AllowedCaveat) *core.AllowedRelation {
+	return &core.AllowedRelation{
+		Namespace: namespaceName,
+		RelationOrWildcard: &core.AllowedRelation_Relation{
+			Relation: relationName,
+		},
+		RequiredCaveat: withCaveat,
+		Deprecation:    deprecation,
+	}
+}
+
+// AllowedRelationWithDeprecationAndExpiration creates a relation reference to an allowed relation.
+func AllowedRelationWithDeprecationAndExpiration(namespaceName string, relationName string, deprecation *core.Deprecation) *core.AllowedRelation {
+	return &core.AllowedRelation{
+		Namespace: namespaceName,
+		RelationOrWildcard: &core.AllowedRelation_Relation{
+			Relation: relationName,
+		},
+		RequiredExpiration: &core.ExpirationTrait{},
+		Deprecation:        deprecation,
+	}
+}
+
+// AllowedRelationWithDeprecationCaveatAndExpiration creates a relation reference to an allowed relation.
+func AllowedRelationWithDeprecationCaveatAndExpiration(namespaceName string, relationName string, withCaveat *core.AllowedCaveat, deprecation *core.Deprecation) *core.AllowedRelation {
```

Comment:
might be easier to add a `WithDeprecation` that takes in an AllowedRelation, adds it, and returns a new one